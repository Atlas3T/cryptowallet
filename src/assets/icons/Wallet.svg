<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
