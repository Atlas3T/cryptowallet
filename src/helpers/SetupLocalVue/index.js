import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
