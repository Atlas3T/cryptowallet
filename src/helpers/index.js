import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
