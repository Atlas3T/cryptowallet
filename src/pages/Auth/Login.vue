<template>
  <div>
    <pin-pad />
    <q-btn
      :label="$t('unlock')"
      style="color: goldenrod;"
      outline
      @click="attemptUnlock"
    />
  </div>
</template>

<script>
import PinPad from '../../components/Auth/PinPad.vue';

export default {

  components: {
    PinPad,
  },

  data() {
    return {
      pin: [],
    };
  },

  computed: {

  },

  beforeCreate() {

  },

  created() {

  },

  beforeMount() {
    this.pinInputListener();
  },

  mounted() {

  },

  beforeUpdate() {

  },

  updated() {

  },

  beforeDestroy() {

  },

  destroyed() {

  },

  methods: {

    /**
     * adds or removes pin input event to pin arr.
     */
    pinInputListener() {
      this.$root.$on('inputPin', (pinArr) => {
        this.pin = pinArr;
      });
    },

    /**
     * @TODO document james.
     */
    attemptUnlock() {
      // @TODO at moment hard coded to first account,
      // need ability to choose which account you're login into.
      const accounts = this.$store.getters['entities/account/query']().get();
      if (this.$acmwcrypto.bcryptCompareString(this.pin.join(''), accounts[0].pinHash) === true) {
        this.$router.push({ path: '/wallet/balance' });
      }
    },
  },

};

</script>

<style scoped>

</style>
