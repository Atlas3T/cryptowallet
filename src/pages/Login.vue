<template>
  <div>
    <pin-pad/>
  </div>
</template>

<script>
import bcrypt from 'bcryptjs';
import PinPad from '../components/Auth/PinPad.vue';

export default {

  components: {
    PinPad,
  },

  data() {
    return {
      pin: '',
    };
  },

  beforeCreate() {

  },

  created() {

  },

  beforeMount() {

  },

  mounted() {

  },

  beforeUpdate() {

  },

  updated() {

  },

  beforeDestroy() {

  },

  destroyed() {

  },

  methods: {
    hashPin(plainPin) {
      const hash = bcrypt.hashSync(plainPin, this.salt);
      console.log(hash);
    },
  },

};

</script>

<style scoped>

</style>
