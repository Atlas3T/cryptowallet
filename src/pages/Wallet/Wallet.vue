<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
