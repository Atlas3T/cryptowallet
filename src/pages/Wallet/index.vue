<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
