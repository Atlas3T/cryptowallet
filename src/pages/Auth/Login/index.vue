<template>
  <div class="center-content-wrapper">
    <div v-if="account">
      <h1 class="account-name-h1 setup">
        {{ account.name }}
      </h1>
      <PinPad mode="auth" />
    </div>
  </div>
</template>

<script>
import { debounce } from 'quasar';
import AES from 'crypto-js/aes';
import encUTF8 from 'crypto-js/enc-utf8';
import { mapState } from 'vuex';
import PinPad from '@/components/Auth/PinPad';
import Wallet from '@/store/wallet/entities/wallet';

export default {

  components: {
    PinPad,
  },

  data() {
    return {
      pin: [],
    };
  },

  computed: {
    ...mapState({
      selectedAccount: (state) => { return state.settings.selectedAccount; },
    }),

    accounts() {
      return this.$store.getters['entities/account/query']().get();
    },

    /**
     * Computes the currently selected account
     */
    account() {
      if (this.selectedAccount) {
        return this.accounts.find((account) => { return account.name === this.selectedAccount; });
      }

      return this.accounts.find((account) => { return account.default; });
    },
  },

  beforeMount() {
    this.pinInputListener();
  },

  /**
   * If there is no account seleted, selects te default one
   */
  mounted() {
    if (!this.selectedAccount) {
      const defaultAccount = this.accounts.find((account) => { return account.default; });
      if (defaultAccount) { this.$store.dispatch('settings/setSelectedAccount', defaultAccount.name); }
    }
  },

  methods: {
    resetPin() {
      this.pin = [];
    },
    /**
     * Adds or removes pin input event to pin arr.
     */
    pinInputListener() {
      this.$root.$on('inputPin', (pin) => {
        this.pin.push(pin);
      });
    },

    /**
     * Compares bcrypt pin string to try and unlock an account
     */
    async attemptUnlock() {
      const minPinLength = 6;
      if (this.pin.length < minPinLength) { return false; }

      try {
        if (this.$CWCrypto.bcryptCompareString(this.pin.join(''), this.account.pinHash) === true) {
          this.$store.dispatch('settings/setLoading', true);
          this.$store.dispatch('settings/setAuthenticatedAccount', this.account.id);
          this.$i18n.locale = this.account.locale;
          await this.decryptData(this.account.id, this.pin.join(''));
          await this.initializeWallets(this.account.id);
          this.$router.push({ path: '/wallet' });
          this.$store.dispatch('settings/setLayout', 'light');

          setTimeout(() => {
            this.$store.dispatch('settings/setLoading', false);
          }, 1000);
        }
      } catch (err) {
        this.errorHandler(err);
      }

      return false;
    },

    debouncedUnlock: debounce(function callback() {
      this.attemptUnlock();
    }, 500),

    /**
     * Decrypts and returns a piece of data
     * @param  {Uint8Array} data
     * @param  {String} password
     * @return {Any}
     */
    decrypt(data, password) {
      const bytes = AES.decrypt(data, password);
      return JSON.parse(bytes.toString(encUTF8));
    },

    async decryptData(id, pass) {
      this.encryptedModels.forEach((model) => {
        if (model.name === 'Account') {
          const result = model.query().where('id', id).get();
          result.forEach((item) => {
            model.AES.forEach((key) => {
              item[key] = this.decrypt(item[key], pass);
            });
          });
        }
        if (model.name === 'Wallet') {
          const result = model.query().where('account_id', id).get();
          result.forEach((item) => {
            model.AES.forEach((key) => {
              if (item[key]) {
                item[key] = this.decrypt(item[key], pass);
              }
            });
          });
        }
      });
    },

    initializeWallets(accountId) {
      return new Promise((resolve) => {
        setTimeout(async () => {
          const initializedWallets = {};
          this.activeWallets[accountId] = initializedWallets;

          const wallets = Wallet.query()
            .where('account_id', accountId)
            .where('enabled', true)
            .get();
          if (wallets.length === 0) {
            resolve();
            return false;
          }

          async function generate(wallet) {
            initializedWallets[wallet.name] = wallet.hdWallet;
          }
          async function generateErc20(wallet) {
            initializedWallets[wallet.name] = wallet.erc20Wallet;
          }

          const erc20Promises = [];
          const promises = [];
          wallets.forEach((wallet) => {
            if (wallet.sdk !== 'ERC20') {
              promises.push(new Promise(async (res) => {
                await generate(wallet);
                res();
              }));
            }
          });
          await Promise.all(promises);

          wallets.forEach((wallet) => {
            if (wallet.sdk === 'ERC20') {
              erc20Promises.push(new Promise(async (res) => {
                const parentSDK = this.coinSDKS[wallet.parentSdk];
                const parentWallet = initializedWallets[wallet.parentName];
                const keyPair = parentSDK.generateKeyPair(parentWallet, 0);
                await generateErc20(keyPair, wallet);
                res();
              }));
            }
          });

          await Promise.all(erc20Promises);

          resolve();
          return false;
        }, 50);
      });
    },

    setAccountModalOpened(value) {
      this.selectAccountsModalOpened = value;
    },
  },
};

</script>

<style scoped>
.account-name-h1 {
  font-size: 2.5em;
  text-align: center;
  padding: 1.5rem;
  padding-top: 1rem;
}
</style>
