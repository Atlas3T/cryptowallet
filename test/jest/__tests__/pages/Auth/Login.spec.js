/* eslint-disable no-magic-numbers */
import { shallowMount } from '@vue/test-utils';
import Login from '@/pages/Auth/Login';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Account from '@/store/wallet/entities/account';
import Wallet from '@/store/wallet/entities/wallet';

describe('Login.vue', () => {
  let wrapper;
  let BackEndService;
  let backEndService;
  let errorHandler;
  let storeMocks;
  let router;
  let $toast;
  const propsData = {};
  const activeWallets = {};
  const mockAccount = {
    $id: 1,
    id: 1,
    uid: 'eaeb9367-442e-8c57-7d98-73ed3840e149',
    refresh_token: 'U2FsdGVkX180Vyx3OSXXJzhifc2LQf6wKqd+KLN/vWYJT0SJ89lo8r3RzW1bXiC1xFz4KL/UwJ9x5LbYfoQvVj8dXWZfDX7HGo9IK0QBMCd0stkMGOpaARxISBcece2YD/4JtOsu9FudP1LYEH2G6mhFSvgtaSkLDgp9f4ZslVrk7Vd7FxEgNGonEl4+bqhBqhKncCCWdWKrYYqnnHFf2GjT/eExe+skkGRRgYLmE0yFBgfHn3BHooF601rfpV5WYwQA7z2rHpnuZsQoULZslg==',
    salt: '$2a$10$3rMsoVY/7P58fLB04dIa8e',
    pinHash: '$2a$10$jqWXqHKd9uRxKo7.zfzQf.CTXCdbhlID0ZZbumux1r7sI.yUOxXsC',
    name: 'Stephen',
    locale: 'en-gb',
    currency: '',
    node: null,
    default: true,
  };
  const walletData = JSON.parse('{"$id":1,"id":1,"account_id":1,"name":"Litecoin","displayName":"Litecoin","symbol":"LTC","sdk":"Bitcoin","network":"LITECOIN_TESTNET","internalChainAddressIndex":4,"externalChainAddressIndex":2,"externalAddress":"QXEjChFGFsGPn9XWYEzCuCTN7Ujsi2j3va","confirmedBalance":10.42288029,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"","parentSdk":"","contractAddress":"","decimals":"","hdWallet":"U2FsdGVkX18tI5wGYddk8+bwlJOR4L/PnY2vZtGNued5xmhdqG0XGivIdnDfmqyrOXdM2kp8vZaxTXUx6I1MXlbnTuCSEGQb+nOR1Z6tBjHfyOCjZl+MSPHqDNrhy3+MU5+NVyfi69ZL0ZPE8xODmgcJyMzM/jQamMoH9F9Qlt6ckwWejrAUYsb7VsRAS2hnVk/3m2QB1a9zre+4vRts6OsqJOVbPO4v6tFORt8H0wh8sIuuVXvjsR3xOj7yzUjVugT6F1YV4F/aCAdmFL4IoKB/i69vG9f7K1UB251pUTnyM+CyuVVNmvpSwlJcQRBHUYkLQrbaDpHVmh3Ey40fDpii34i+T8FFVhf7XOlsdz9Wzjof0hBARlBfwYlRy9lbuoIJ97kKVYbIA0b1f2SvXGnGGLEOgcEjUiQlzzl7atLv7vFZzcB9PFqDgxB7OhKXqz05/uwmtaEQ6roi/g3VDdXE0h7+g1RhzbqiYfZeofOHJ3ENtve6Xcznx9YpxEcWirjPo4XZ6cYDTFHCd0X2wz7dYFClV1ADgwlUqZ2VzP2BlYYKTKfZbfLZpdH968ZtOxXhCmVsIw14FiAmsUhdq5XN7p8xiWIqU7QnucuRE32upUwCN17EtWborSArvh6ZmjNPnkwyf/BA2/LAmp3/2Xj2D+8GnpOe/8WFYxmXrAYdSjG/Hxytm8aOvP4hcsi1gMzCnfw1ZX9EqxJsKAfKenpBbN6mfsVP7osRLPzU6aCwnS/hYPM+UIY0bUqZDiCtHbY4jNXRgcw44zPDYe7+PTmO12mKzD0h7w3TEBH8dRA80XscYKwLnbLsxPhK8Cjr7kVU31uSIKcG9X4mBLtYHxtWjqZINuWuqFajYXhHIDmlnno6ieRkLy1C1AN0PuG51WlvpSPqFNvVB26IHhhZzRC0aUjIXwWjCuEBT88y0IsDKCh1jRYwFcxjzv5MkiSU8OCNi79GbRek5cvvZ3QbkI8p3PCr691a+Mcd39CKa7pjzhUJFZmpUPusjeE2s2R2s+RF5O3ufDrdhRYE/YItXVhdsplwzeeEIZro/SHwCLSBx7pILZunfQ6lLe8c32eRtoZilkMZs7Svip/lWzVbfLllv5Tcf/adqDdgpaXZujiaTJ/se4dlYBL129YW0o/0Gv6h5Xg704WbJTk9DSSvwfwxhm8v612uevHU8Gem7XGLD6/xmdCJHQBTfrgKiw7nkIQex/u6UhWxrMiHfJz1MUTModhIEhWGGbhQTYEA7CQ=","erc20Wallet":""}');
  const erc20WalletData = JSON.parse('{"$id":5,"id":5,"account_id":1,"name":"Catalyst","displayName":"Catalyst","symbol":"CAT","sdk":"ERC20","network":"ETHEREUM_ROPSTEN","internalChainAddressIndex":0,"externalChainAddressIndex":0,"externalAddress":"0x5d7DE0a3f75E0e9c3163413870938bEE612e07D6","confirmedBalance":500,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"Ethereum","parentSdk":"Ethereum","contractAddress":"0x26705403968a8c73656a2fed0f89245698718f3f","decimals":3,"hdWallet":"","erc20Wallet":{"decimals":3,"address":"0x5d7DE0a3f75E0e9c3163413870938bEE612e07D6","network":{"name":"ETHEREUM_ROPSTEN","networkName":"ropsten","bip":60,"getTranApi":"http://api-ropsten.etherscan.io/api?module=account&action=txlist&address=","getBalanceApi":"https://api-ropsten.etherscan.io/api?module=account&action=balance&address=","getErc20TranApi":"http://api-ropsten.etherscan.io/api?module=account&action=tokentx&contractaddress=","feeApi":"https://api.blockcypher.com/v1/eth/main","provider":"https://ropsten.infura.io/v3/352fc30cd8364caabaea4a3d67da773f","chainId":3},"name":"Catalyst","symbol":"CAT","contract":"0x26705403968a8c73656a2fed0f89245698718f3f"}}');
  const customStore = {
    state: {
      settings: {
        selectedAccount: 'Stephen',
      },
    },
  };
  function wrapperInit(options) {
    return shallowMount(Login, options);
  }

  function storeInit(custom) {
    $toast = {
      create: jest.fn(),
    };
    errorHandler = jest.fn();
    backEndService = {
      connect: jest.fn().mockResolvedValue(true),
      loadPriceFeed: jest.fn(),
    };
    BackEndService = jest.fn().mockImplementation(() => {
      return {
        connect: jest.fn(),
        loadPriceFeed: jest.fn(),
      };
    });
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    Account.$insert({ data: [mockAccount] });
    Wallet.insert({ data: [walletData, erc20WalletData] });
    router.push({ path: '/' });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      propsData,
      store: storeMocks.store,
      mocks: {
        BackEndService,
        backEndService,
        errorHandler,
        activeWallets,
        $toast,
      },
    });
  }

  beforeEach(() => { storeInit(); });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the account name', () => {
    expect(wrapper.text()).toEqual('Stephen');
  });

  it('resets the pin', () => {
    wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
    wrapper.vm.resetPin();
    expect(wrapper.vm.pin).toEqual([]);
  });

  it('listens for pin input', () => {
    wrapper.vm.$root.$emit('inputPin', 1);
    wrapper.vm.$root.$emit('inputPin', 2);
    wrapper.vm.$root.$emit('inputPin', 3);
    wrapper.vm.$root.$emit('inputPin', 4);

    expect(wrapper.vm.pin).toEqual([1, 2, 3, 4]);
  });

  it('sets the authenticated account in the store on successful login', (done) => {
    wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
    wrapper.vm.attemptUnlock();
    setTimeout(() => {
      expect(storeMocks.actions.setLoading).toHaveBeenCalled();
      expect(storeMocks.actions.setAuthenticatedAccount).toHaveBeenCalled();
      done();
    }, 1000);
  });

  it('decrypts the wallet data and initialises them', (done) => {
    const wallets = Wallet.all();
    wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
    wrapper.vm.attemptUnlock();
    setTimeout(() => {
      expect(wallets[0].hdWallet.ext.xpub).toEqual('xpub6Dswyp5Ezb28B7AG7yWDCS7QLDa5kPAgt6mHYQgw11krarWYviyKwRcYNbeutQRsNtQn1DD9SnaXHMpwpzAEDnBxvZZrDFQ2ehUGw8kxH6u');
      expect(wrapper.vm.activeWallets[1].Litecoin).toBeTruthy();
      expect(wrapper.vm.activeWallets[1].Catalyst).toBeTruthy();
      done();
    }, 1000);
  });

  it('creates a backEndService and loads the price feed', (done) => {
    wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
    wrapper.vm.attemptUnlock();
    setTimeout(() => {
      expect(BackEndService).toHaveBeenCalled();
      expect(backEndService.connect).toHaveBeenCalled();
      expect(backEndService.loadPriceFeed).toHaveBeenCalled();
      done();
    }, 1000);
  });

  it('displays a wrong pin toast if pin doesn\'t match', (done) => {
    const mockResetPin = jest.fn();
    wrapper.setMethods({ resetPin: mockResetPin });

    const mockResetState = jest.fn();
    wrapper.vm.$refs.PinPad = {
      resetState: mockResetState,
    };

    wrapper.vm.pin = [0, 0, 0, 0, 0, 1];
    wrapper.vm.attemptUnlock();
    setTimeout(() => {
      expect(wrapper.vm.$toast.create).toHaveBeenCalled();
      expect(mockResetState).toHaveBeenCalled();
      expect(mockResetPin).toHaveBeenCalled();
      done();
    }, 500);
  });

  it('catches errors', (done) => {
    mockAccount.id = 2;
    mockAccount.default = false;
    storeInit(customStore);
    wrapper.vm.pin = [0, 0, 0, 0, 0, 0];
    wrapper.vm.attemptUnlock();
    backEndService.connect.mockImplementationOnce(() => { throw new Error('Test Error'); });
    setTimeout(() => {
      expect(errorHandler).toHaveBeenCalled();
      done();
    }, 1000);
  });
});
