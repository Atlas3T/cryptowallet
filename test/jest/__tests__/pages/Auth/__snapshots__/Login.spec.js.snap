// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders and matches snapshot 1`] = `
<div>
  <div
    class="center-content-wrapper"
  >
    <div>
      <h1
        class="account-name-h1 setup"
      >
        
        Stephen
      
      </h1>
       
      <pinpad-stub
        mode="auth"
      />
    </div>
  </div>
</div>
`;
